lua/bufferline/highlight.lua

```diff
@@ -23,97 +20,91 @@ local function color(index, groups, default, default_cterm)
-  --     Inactive: invisible but not current buffer
-  --        -Icon: filetype icon
-  --       -Index: buffer index
-  --         -Mod: when modified
-  --        -Sign: the separator between buffers
-  --      -Target: letter in buffer-picking mode
-  local hl = vim.go.termguicolors and
-    function(group, background, foreground, bold)
-      vim.api.nvim_set_hl(0, group, {
-        bg = background,
-        bold = bold,
-        default = true,
-        fg = foreground,
-      })
-    end or
-    function(group, background, foreground, bold)
-      vim.api.nvim_set_hl(0, group, {
-        bold = bold,
-        ctermbg = background,
-        ctermfg = foreground,
-        default = true,
-      })
-    end
-
-  hl('BufferCurrent',        bg_current, fg_current)
-  hl('BufferCurrentIndex',   bg_current, fg_special)
-  hl('BufferCurrentMod',     bg_current, fg_modified)
-  hl('BufferCurrentSign',    bg_current, fg_special)
-  hl('BufferCurrentTarget',  bg_current, fg_target, true)
-  hl('BufferInactive',       bg_inactive, fg_inactive)
-  hl('BufferInactiveIndex',  bg_inactive, fg_subtle)
-  hl('BufferInactiveMod',    bg_inactive, fg_modified)
-  hl('BufferInactiveSign',   bg_inactive, fg_subtle)
-  hl('BufferInactiveTarget', bg_inactive, fg_target, true)
-  hl('BufferTabpageFill',    bg_inactive, fg_inactive)
-  hl('BufferTabpages',       bg_inactive, fg_special, true)
-  hl('BufferVisible',        bg_visible, fg_visible)
-  hl('BufferVisibleIndex',   bg_visible, fg_visible)
-  hl('BufferVisibleMod',     bg_visible, fg_modified)
-  hl('BufferVisibleSign',    bg_visible, fg_visible)
-  hl('BufferVisibleTarget',  bg_visible, fg_target, true)
-
-  vim.api.nvim_set_hl(0, 'BufferCurrentIcon', {default = true, link = 'BufferCurrent'})
-  vim.api.nvim_set_hl(0, 'BufferInactiveIcon', {default = true, link = 'BufferInactive'})
-  vim.api.nvim_set_hl(0, 'BufferVisibleIcon', {default = true, link = 'BufferVisible'})
-  vim.api.nvim_set_hl(0, 'BufferOffset', {default = true, link = 'BufferTabpageFill'})
+--- Generate a foreground color.
+--- @param groups table<string> the groups to source the foreground color from.
+--- @param default string the foreground color to use if no `groups` have a valid foreground color.
+--- @param default_cterm nil|number|string the color to use if no `groups` have a valid color and `termguicolors == false`.
+--- @return number|string color
+local function fg(groups, default, default_cterm)
+  return color('foreground', groups, default, default_cterm or default)
+end
 
-  icons.set_highlights()
+--- Set the default highlight `group_name` as a link to `link_name`
+--- @param group_name string the name of the group to by-default be linked to `link_name`
+--- @param link_name string the name of the group to by-default link `group_name` to
+local function set_default_hl_link(group_name, link_name)
+  set_hl(0, group_name, {default = true, link = link_name})
 end
 
-return highlight
+return {
```

Comment:
Since we're just returning one method (`highlight.setup`) I figured I'd just return `{setup = …}` instead of all the:

```lua
--- @class bufferline.highlight
local highlight = {}

function highlight.setup()
  …
end

return highlight
```